src/server/services/registrar.js

```diff
@@ -164,8 +164,61 @@ async function getShortname(offeredTermCode, classSectionID) {
   }
 }
 
+/**
+ * Gets the week number, given an academic term and a date
+ *
+ * @param {string} term Academic term, formatted as YYQ (e.g. 20S)
+ * @param {string} date Date string formatted as MM/DD/YYYY
+ * @returns {?string} Returns week number if date is valid in term
+ */
+async function getWeekNumber(term, date) {
+  try {
+    const response = await registrar.call({
+      url: `sis/api/v1/Dictionary/TermSessionsByWeek?SessionTermCode=${term}&SessionCode=RG&PageSize=12`,
```

Comment:
@chern I am not sure why you have PageSize=12. Also, when calling registar.call() you don't need to put the GET parameters in the URL. You should be able to pass it as a param, like:

```
const response = await registrar.call({
      url: 'sis/api/v1/Dictionary/TermSessionsByWeek',
      params: {
          SessionTermCode: term,
          SessionCode: RG
      }
});
```